packages/core/src/errors.ts

```diff
@@ -22,7 +22,8 @@ export enum SerializationFailure {
   InvalidType = 'INVALID_TYPE',
   Overflow = 'OVERFLOW',
   InvalidAddress = 'INVALID_ADDRESS',
-  MaxLengthLimit = 'MAX_LENGTH_LIMIT'
+  MaxLengthLimit = 'MAX_LENGTH_LIMIT',
+  InvalidScriptType = 'INVALID_SCRIPT_TYPE'
```

Comment:
Seeing as there'll be support of other (non-simple/native) scripts types in future, it would make sense to make this more descriptive. E.g. `INVALID_SIMPLE_SCRIPT_TYPE`

Reply:
Thanks for the input @james-iohk I was totally oblivious to the fact these are called Simple Scripts, I have been referring to those as Native Scripts since forever (the Cardano serialization lib calls them as such (https://input-output-hk.github.io/cardano-js-sdk/classes/_cardano_sdk_core.CSL.NativeScript.html), however looking at the reference documentation (https://github.com/input-output-hk/cardano-node/blob/master/doc/reference/simple-scripts.md), they are indeed referred as simple scripts. I will refactor the name to SimpleScript instead of NativeScript. What do you think @mkazlauskas?

Reply:
Agree, let's rename :+1: Thanks @james-iohk

Reply:
addressed here https://github.com/input-output-hk/cardano-js-sdk/pull/377/commits/b5ac8f5732130918aabc58ba3243252df838b4a4

Reply:
@james-iohk @mkazlauskas I already made the refactor; however, I was checking the CDDL schema of transactions for another ticket I am working on, and I see that in the Alonzo CDDL schema this type of scripts are referred to as Native Scripts formally: https://github.com/input-output-hk/cardano-ledger/blob/b44a3b8959c00e8243c3004c01adf412d1b2d317/eras/alonzo/test-suite/cddl-files/alonzo.cddl#L351

I think the schema should be followed over anything else no?, le me know your thoughts.

Reply:
cc @rhyslbw

Reply:
Yes, and it makes sense to call these scripts "native", given Plutus is an explicit type: https://github.com/input-output-hk/cardano-ledger/blob/b44a3b8959c00e8243c3004c01adf412d1b2d317/eras/alonzo/test-suite/cddl-files/alonzo.cddl#L274

Reply:
Let's base on the Ogmios [representation](https://github.com/CardanoSolutions/ogmios/blob/e2a6ff63e06be3b16065c82a9c0e3be7d51dfca1/docs/static/ogmios.wsp.json#L3910-L3937), where they can be either `Native`, `PlutusV1`, or `PlutusV2`: 

``` json
  , "Script":
    { "oneOf":
      [ { "type": "object"
        , "title": "Native"
        , "additionalProperties": false
        , "required": [ "native" ]
        , "properties":
          { "native": { "$ref": "#/definitions/Script[Native]" }
          }
        }
      , { "type": "object"
        , "title": "PlutusV1"
        , "additionalProperties": false
        , "required": [ "plutus:v1" ]
        , "properties":
          { "plutus:v1": { "$ref": "#/definitions/Script[Plutus]" }
          }
        }
      , { "type": "object"
        , "title": "PlutusV2"
        , "additionalProperties": false
        , "required": [ "plutus:v2" ]
        , "properties":
          { "plutus:v2": { "$ref": "#/definitions/Script[Plutus]" }
          }
        }
      ]
    }
```

Reply:
addressed here: https://github.com/input-output-hk/cardano-js-sdk/pull/377/commits/70cbbbcdb8d9a3cb1af9edd313274d5e1ace0dff